### ISSUE TITLE
check whether rust has overflow turned on by default in the contract

### ISSUE THREAD

**Issue by goastler** [prosopo/protocol#194]

**Comment by goastler** [comment]
Just tested it. Rust has compile time overflow detection, e.g.
```rust
let mut a: u8 = 255;
a += 1;
a += 1; // overflow err
```

However, if you pass it in as a parameter then it cannot detect overflows, e.g.
```rust

        #[ink(message)]
        pub fn abc(&self, b: u8) -> u8 {
            let mut a: u8 = 255;
            a + b // overflows during runtime, no err
        }
```

This is something to be conscious of during dev.

**Comment by goastler** [comment]
Further to this issue, overflow checks can be enabled during runtime by setting a compiler flag. However, the wasm validation fails when `overflow-checks = true` in the Cargo.toml because the panic thrown by the overflow checks is not supported in wasm. At the moment, cargo contract recommends disabling overflow-checks until further development has been made in handling these panics.

tldr; come back to this when they (ink) have fixed it
